lib/logstash/inputs/gelf.rb

```diff
@@ -157,7 +159,7 @@ def self.from_json_parse(json)
   # legacy_parse uses the LogStash::Json class to deserialize json
   def self.legacy_parse(json)
     o = LogStash::Json.load(json)
-    LogStash::Event.new(o) if o
+    LogStash::Event.new(o)
```

Comment:
I think there is? I'd just make sure that if `LogStash::Json.load(json)` returns `nil` that it will trickle correctly and simply hit the `next if xxxx.nil?` in the listner? seems easy to verify and fix it's not the case now?